.github/workflows/backwards_compatibility_marqo_orchestrator.yml

```diff
@@ -0,0 +1,174 @@
+# Orchestrator workflow
+name: Marqo Compatibility Tests Orchestrator
+
+on:
+  push:
+    branches:
+      - mainline
+    paths-ignore:
+      - '**.md'
+  workflow_dispatch:
+    inputs:
+      to_version:
+        description: 'Target Marqo version'
+        required: true
+      to_image:
+        description: 'Target Marqo image (optional)'
+        required: false
+      max_versions_to_test:
+        description: 'Max versions to test'
+        required: false
+
+# Setting MAX_VERSIONS_TO_TEST, this can be a configurable value or if no input is provided, it can be a default value.
+env:
+  MAX_VERSIONS_TO_TEST: ${{ github.event.inputs.max_versions_to_test || 3 }}
+
+jobs:
+  invoke-build-push-img-github-workflow:
+    # Responsible for invoking build_push_img.yml GitHub actions workflow in the same repo
+    # that builds a marqo commit using the latest commit and uploads it to a private ECR Registry.
+    # More in /marqo/.github/workflows/build_push_img.yml
+    name: Invoke build_push_img.yml github actions workflow
+    runs-on: ubuntu-latest
+    environment: marqo-test-suite
+    steps:
+      - name: Checkout marqo repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      # step to check for image existence - it uses aws cli to check if the image exists in the ECR registry "marqo-compatibility-tests"
+      - name: check for image existence
+        id: check-image
+        run: |
+          echo "Checking for image existence"
+          if aws ecr describe-images --repository-name marqo-compatibility-tests --image-ids imageTag=${{ github.sha }}; then
+            echo "::set-output name=image_exists::true"
+          else
+            echo "::set-output name=image_exists::false"
+          fi
+
+      # Step to build and push marqo image to ECR, only executes if the image does not exist signified by image_exists == false.
```

Comment:
we already have this action merged in. We can trigger that action here.

Reply:
Also, consider using GH action like https://github.com/marketplace/actions/workflow-dispatch-and-wait to simplify the process. We need to have a brief look to make sure these 3rd-party actions do not contain malicious code, and pin to a specific git commit sha.

Reply:
Removed this logic of invoking actions using github API altogether and started using jobs with `uses: ./.github/workflows/build_push_img.yml` everywhere